lib/cms/hippo_importer/base.rb

```diff
@@ -0,0 +1,99 @@
+module Cms
+  module HippoImporter
+    class Base
+      def initialize(data:, docs:, to: 'en', parser: HippoXmlParser, html_decoder: HTMLEntities.new)
+        @records = parser.parse(data, [hippo_type])
+        @_site = to || 'en'
+        @docs = docs || []
+        @html_decoder = html_decoder
+      end
+
+      attr_reader :records, :html_decoder
+
+      def site
+        @site ||= Comfy::Cms::Site.find_by(label: @_site)
+      end
+
+      def hippo_type
+        'contentauthoringwebsite:Guide'
+      end
+
+      def layout
+        @_layout ||= (site.try(:layouts) || []).first
+      end
+
+      def parent
+        @_parent ||= Comfy::Cms::Page.where(parent_id: nil, site: site).first
+      end
+
+      def decoded(str)
+        decoded = html_decoder.decode(str)
+        html = Nokogiri::HTML.parse(decoded)
+        remove_unused_elements(html)
+        convert_videos(html)
+        ReverseMarkdown.site = site
+        ReverseMarkdown.convert(html)
+      end
+
+      def remove_unused_elements(html)
+        ['//p[@class="intro"]/img', '//a[@class="action-email"]',
+         '//form[@class="action-form"]', '//span[@class="collapse"]'].each do |path|
+          html.xpath(path).remove
+        end
+      end
+
+      def convert_videos(html)
+        html.xpath('//iframe[starts-with(@src, "https://www.youtube.com/embed")]').each  do |e|
+          e.replace('({' + e.attributes['src'].value.gsub('https://www.youtube.com/embed/', '') + '})')
+        end
+      end
+
+      def docs
+        @cached_docs ||= @docs.empty? ? Comfy::Cms::Page.all.map(&:slug) : @docs
+      end
+
+      def next_docs
+        @next_docs ||= @_site == 'en' ? docs : docs.map { |a| Comfy::Cms::Page.find_by(slug: a) }.map(&:translation_id)
```

Comment:
The `@_site == 'en'` comparison is done in enough places I'd move it into a predicate, perhaps `english?`